R/TealAppDriver.R

```diff
@@ -45,25 +46,51 @@ TealAppDriver <- R6::R6Class( # nolint: object_name.
     initialize = function(data,
                           modules,
                           filter = teal_slices(),
-                          title = build_app_title(),
+                          title_args = list(),
                           header = tags$p(),
                           footer = tags$p(),
-                          landing_popup = NULL,
+                          landing_popup_args = NULL,
                           timeout = rlang::missing_arg(),
                           load_timeout = rlang::missing_arg(),
                           ...) {
       private$data <- data
       private$modules <- modules
       private$filter <- filter
+
+      new_title <- modifyList(
+        list(
+          title = "Custom Teal App Title",
+          favicon = "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/teal.png"
```

Comment:
Hey @vedhav sorry for commenting in already closed PR but I do have a few comments. Please open a new issue / PR if you think it's relevant and you agree with me.

Can we use the logo file from the installed teal package to avoid url requests? This is under assumption that the logo is already there (haven't checked)

Reply:
Thanks for the comments @pawelru. I will address them first thing on Monday, most of them seem reasonable.

Reply:
I'm having second thoughts about using a static logo from the package path. In order to achieve this we need to do these two things:
1. Add the favicon to the inst, not a big deal as we can reduce its size.
2. Serve this file as static content ([addResourcePath](https://shiny.posit.co/r/reference/shiny/1.0.1/addresourcepath.html)) so it can be accessed by shiny.
It is the second point where I have the trouble, I do not feel that serving static content just for a favicon is a good idea.
I think it's safe to use this remote URL as it is controlled by us.

Reply:
I'm not a fan of addResourcePath for favicon neither. I think URL is the best solution

Reply:
I suggest to revisit this once again as a separate task. We already have lived with urls for a while so it's not a big issue. It's just an idea for performance enhancement - not a bug per se.